File: kangaroo_business_app_sdk_android/android/src/main/kotlin/com/kangaroo/flutterplugin/android/base/PluginChannelHandler.kt
package com.kangaroo.flutterplugin.android.base

import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodCall

interface PluginChannelHandler {
    val methodChannel: String
    val eventChannel: String
    suspend fun onMethodCall(call: MethodCall): Any?
    fun getStreamHandler(): EventChannel.StreamHandler
}

File: kangaroo_business_app_sdk_ios/ios/Classes/base/PluginChannelHandler.swift
//
//  PluginChannelHandler.swift
//  fluttertoast
//
//  Created by Paul on 2021-10-17.
//

import Foundation
import Flutter

protocol PluginChannelHandler {
    var methodChannel: String { get set}
    var eventChannel: String { get set}

    func onMethodCall(call: FlutterMethodCall) async -> Any?
    
    func getStreamHandler() -> (FlutterStreamHandler & NSObjectProtocol)?
}
